pkg/bindings/containers/types_resizetty_options.go

```diff
@@ -43,33 +43,19 @@ func (o *ResizeTTYOptions) ToParams() (url.Values, error) {
 		if reflect.Ptr == f.Kind() {
 			f = f.Elem()
 		}
-		switch f.Kind() {
-		case reflect.Bool:
-			params.Set(fieldName, strconv.FormatBool(f.Bool()))
-		case reflect.String:
-			params.Set(fieldName, f.String())
-		case reflect.Int, reflect.Int64:
-			// f.Int() is always an int64
-			params.Set(fieldName, strconv.FormatInt(f.Int(), 10))
-		case reflect.Uint, reflect.Uint64:
-			// f.Uint() is always an uint64
-			params.Set(fieldName, strconv.FormatUint(f.Uint(), 10))
-		case reflect.Slice:
-			typ := reflect.TypeOf(f.Interface()).Elem()
-			switch typ.Kind() {
-			case reflect.String:
-				sl := f.Slice(0, f.Len())
-				s, ok := sl.Interface().([]string)
-				if !ok {
-					return nil, errors.New("failed to convert to string slice")
+		switch {
+		case util.IsSimpleType(f):
+			params.Set(fieldName, util.SimpleTypeToParam(f))
+		case f.Kind() == reflect.Slice:
+			for i := 0; i < f.Len(); i++ {
+				elem := f.Index(i)
+				if util.IsSimpleType(elem) {
+					params.Add(fieldName, util.SimpleTypeToParam(elem))
+				} else {
+					return nil, errors.New("slices must contain only simple types")
```

Comment:
A lot of very similar code like this.  It would be good if we could manage to add a param or two to it and just have it in one place.

Reply:
The last comment was for potential future work, not for this PR.

Reply:
Isn't that a generated code?

Reply:
But yeah, it could be improved by using some common interface. Or by generics if they will add them to go :smile:

Reply:
It is generated code, but we are including it in the source, since everyone would have to generate it every time.
Getting to more shared code would be nice.

Reply:
@rhatdan  Another approach would be generate `ToParams()` differently. Now all the `ToParams()` are the same, they use reflection. Maybe with using `ast.*` we could create more performant  non-reflection `ToParams()` specialized for each struct. `WithXYZ()` methods seems to be generated in such a way.

Reply:
If you want to take a stab at that I am sure we would look at it.